server.py

```diff
@@ -43,31 +55,30 @@
         statement = fl.read()
     questions[i] = Question(output=output, statement=statement)
 
+
 @app.route("/")
 def changePath():
     return redirect("/dashboard")
 
+
 @app.get("/dashboard")
 def dashboard():
     return template("dashboard.html", contests=contests)
 
+
 @app.get("/contest/<code>/<number>")
-def contest(code,number):
+def contest(code, number):
     statement = questions[number].statement
     return template("index.html", question_number=number, question=statement)
 
+
 @app.get("/contest/<code>")
 def contest(code):
-    for contest in contests:
-        if(contest.code == code):
-            description = contest.description
-            questions = contest.questions
-    return template("contest.html", code=code, description=description, questions=questions)
-
-@app.get("/question/<number>")
-def question(number):
-    statement = questions[number].statement
-    return template("index.html", question_number=number, question=statement)
+    if not code in contests.keys():
```

Comment:
```suggestion
    if not code in contests:
```

if you use `.keys()` it makes a list-ish item and then looks up in that. Instead use the constant time lookup property of dictionaries